src/ert/scheduler/job.py

```diff
@@ -156,13 +134,37 @@ async def __call__(
         for attempt in range(max_submit):
             await self._submit_and_run_once(sem)
 
-            if self.returncode.cancelled() or (
-                self.returncode.done() and self.returncode.result() == 0
-            ):
+            if self.returncode.cancelled():
+                break
+
+            if self.returncode.result() == 0:
+                callback_status, status_msg = forward_model_ok(self.real.run_arg)
+                if self._callback_status_msg != "":
+                    self._callback_status_msg = status_msg
+                else:
+                    self._callback_status_msg += (
+                        f"\nstatus from done callback: {status_msg}"
+                    )
+
+                if callback_status == LoadStatus.LOAD_SUCCESSFUL:
+                    await self._send(State.COMPLETED)
+                else:
+                    assert callback_status in (
+                        LoadStatus.LOAD_FAILURE,
+                        LoadStatus.TIME_MAP_FAILURE,
+                    )
+                    await self._send(State.FAILED)
                 break
             elif attempt < max_submit - 1:
-                message = f"Realization: {self.iens} failed, resubmitting"
+                message = (
+                    f"Realization {self.iens} failed, "
+                    f"resubmitting for attempt {attempt+2} of {max_submit}"
+                )
                 logger.warning(message)
+                self.returncode = asyncio.Future()
+                self.started = asyncio.Event()
+            else:
```

Comment:
Is this else statement the fix for the problem?

Reply:
The main fix for the problem is to move the calling of `forward_model_ok` out from `_submit_and_run_once` and in to `__call__` to avoid it being called upon every attempt.

Reply:
But before, the calling of `forward_model_ok` was only called upon return_code 0. Aren't we doing the same now?

Reply:
You are right. The problem was that we did  `self._send(State.FAILED)` for every unsuccessful attempt, and this triggered a storage change.

Reply:
This means that we should update our state diagram as retry now goes from running, right?